pages/apply/expect.tsx

```diff
@@ -7,27 +7,90 @@ import Layout from '../../components/layout/resident-layout';
 import DeleteLink from '../../components/delete-link';
 import Announcement from '../../components/announcement';
 import { signOut } from '../../lib/store/cognitoUser';
+import { Applicant } from '../../domain/HousingApi';
+import { CALCULATE_BEDROOMS } from '../../lib/utils/bedroomCalculator';
+import { useState, useEffect } from 'react';
 
 const WhatToExpect = (): JSX.Element => {
   const router = useRouter();
   const dispatch = useAppDispatch();
 
+  const [CalculatedBedrooms, setCalculatedBedrooms] = useState(0);
+  const [waitingTime, setWaitingTime] = useState(0);
+
   const onCancel = () => {
     dispatch(signOut());
     router.push('/');
   };
 
+  const getAge = (date: string) => {
+    let today = new Date();
+    let birthDate = new Date(date);
+    let age = today.getFullYear() - birthDate.getFullYear();
+    let m = today.getMonth() - birthDate.getMonth();
+    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+    return age;
+  };
+
+  const getGender = (gender: string) => {
+    switch (gender) {
+      case 'M':
+        return 'male';
+      case 'F':
+        return 'female';
+    }
+  };
+
+  const getWaitingTime = (bedroom: number) => {
+    switch (bedroom) {
+      case 1:
+        return 4;
+      case 2:
+        return 11;
+      case 3:
+        return 12;
+      case 4:
+        return 17;
+      default:
+        return 17;
+    }
+  };
+
+  const applicants = useAppSelector((store) =>
+    [store.application.mainApplicant, store.application.otherMembers]
+      .filter((v): v is Applicant | Applicant[] => v !== undefined)
+      .flat()
+  );
+
+  const bedroomArray = applicants.map((individual) => {
+    let age = getAge(individual['person']!['dateOfBirth']!);
+    let gender = getGender(individual['person']!['gender']!);
+    let relationship = individual['person']!['relationshipType']!;
+    return [age, gender, relationship];
+  });
+
+  const bedroom_results = CALCULATE_BEDROOMS(bedroomArray);
+  useEffect(() => {
+    setCalculatedBedrooms(bedroom_results);
```

Comment:
You don't want state for this - it's a calculated value so you have two options: 
- calculate it every render
- memoise it with useMemo

Personally I'd just calculate it every render. Unless there's a performance issue that you've spotted it'll keep the code simpler. Don't use state though - you'll introduce an unnecessary render.